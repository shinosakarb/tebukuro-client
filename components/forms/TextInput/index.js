// @flow
import React from 'react'

type Props = {
  id: string,
  value: string | number,
  onChange: Function,
}

export default (props: Props) => (
  <div>
    <label htmlFor={props.id}>
      { props.id }
      <input type="text" {...props} />
    </label>
  </div>
)
