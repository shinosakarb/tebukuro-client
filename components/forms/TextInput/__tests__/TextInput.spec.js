import React from 'react'
import { shallow } from 'enzyme'
import shallowToJson from 'enzyme-to-json'
import TextInput from '../index'

const testProps = {
  id: 'name',
  value: 'event1',
  onChange: jest.fn(),
}

describe('TextInput', () => {
  it('render event form component', () => {
    const wrapper = shallow(<TextInput {...testProps} />)
    const tree = shallowToJson(wrapper)

    expect(tree).toMatchSnapshot()
  })
})
