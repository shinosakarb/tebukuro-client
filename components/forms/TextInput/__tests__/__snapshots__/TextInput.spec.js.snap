// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput render event form component 1`] = `
<div>
  <label>
    <input
      name="name"
      onChange={[Function]}
      type="text"
      value="event1"
    />
  </label>
</div>
`;
